### PR TITLE
Multi-interface support

### LINKED ISSUE #549: Add multiple interface support
It would be useful for Firezone to be able to manage multiple network interfaces with subnets assigned to each.

### PR DESCRIPTION
**Pull request by jamilbk**
Fixes #549 

- [x] Create networks table with proper fields, validations
- [x] Create migration to move config-file settings into first row of `networks` table, add `network_id` to `devices`
- [ ] Add new UI section, forms, modal
- [ ] Update existing device form, fields to be network-aware
- [ ] Generating configs should be network-aware
- [ ] Propagate interface creation, configuration, and deletion to `fz_vpn`
- [ ] Propagate interface creation to `fz_wall`
- [x] Remove old WireGuard interface cruft from Chef recipes / scripts
- [x] Remove WireGuard interface config from config file
- [x] Add appropriate telemetry events
- [ ] Update docs

### CODE REVIEW

**Comment by jamilbk** [comment]
@princemaple This PR still has a ton of work left to do, but the data piece is (mostly) complete. Wanted to get just a really quick sanity check to see what you think.

This PR will essentially move the WireGuard interface config to the app layer and will allow users to create multiple WireGuard subnets, each with their own PrivateKey and associated devices.

File: apps/fz_http/priv/repo/migrations/20220527025915_create_networks.exs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+    # Exclusion index to prevent overlapping CIDR ranges
+    execute("ALTER TABLE networks ADD EXCLUDE USING gist (ipv4_network inet_ops WITH &&)")
+    execute("ALTER TABLE networks ADD EXCLUDE USING gist (ipv6_network inet_ops WITH &&)")
+
+    create unique_index(:networks, [:ipv4_address], where: "ipv4_address IS NOT NULL")
+    create unique_index(:networks, [:ipv6_address], where: "ipv6_address IS NOT NULL")
+    create unique_index(:networks, [:interface_name])
+    create unique_index(:networks, [:private_key])
+    create unique_index(:networks, [:listen_port])
```
Comment (princemaple) [inline apps/fz_http/priv/repo/migrations/20220527025915_create_networks.exs:82]: Do `interface_name` and `listen_port` need to take `site` / `user` into consideration? I probably don't understand the problem we are solving enough.

Comment (princemaple) [reply apps/fz_http/priv/repo/migrations/20220527025915_create_networks.exs:82]: If it's per machine, it's all good here.

**Comment by princemaple** [comment]
What happend in https://github.com/firezone/firezone/commit/83adf0e7e57cad5732ae6a59109a134e75c265dd ? All parameters removed.

File: apps/fz_http/priv/repo/migrations/20220527025915_create_networks.exs (hunk 1 above)
Comment (jamilbk) [reply apps/fz_http/priv/repo/migrations/20220527025915_create_networks.exs:82]: That’s a good point. We don’t have multi site yet but it’s planned. I’ll add a `site_id` field to scope it.

**Pull request closed without merging**